brian2/codegen/runtime/cython_rt/templates/spikegenerator.pyx

```diff
@@ -1,39 +1,45 @@
 {% extends 'common.pyx' %}
 
-{% block template_support_code %}
-cdef int _start_idx = 0
-cdef double _current_time = 0.0
-{% endblock %}
-
 {% block maincode %}
 
-    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time } #}
-     
-    global _start_idx, _current_time
-    
-    # Check if we have been reset, if so reset the cache
-    if t<_current_time:
-        _start_idx = 0
-    
-    _current_time = t
+    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time, period, _lastindex } #}
 
     # TODO: We don't deal with more than one spike per neuron yet

```

Comment:
I guess that would be better than a TODO in the code :) But probably a check before hand, not during the run, right? But this would involve applying the rounding/epsilon logic before. Hmm, maybe we should actually do a conversion from times to time bins before hand?